q2_fragment_insertion/_insertion.py

```diff
@@ -302,3 +303,43 @@ def classify_otus_experimental(
              "i.e. are results from the same 'sepp' run."))
 
     return pd_taxonomy.set_index('Feature ID')
+
+
+def filter_features(table: biom.Table,
+                    tree: NewickFormat) -> (biom.Table, biom.Table):
+
+    # load the insertion tree
+    tree = skbio.TreeNode.read(str(tree))
+    # collect all tips=inserted fragments+reference taxa names
+    fragments_tree = {
+        str(tip.name)
+        for tip in tree.tips()
+        if tip.name is not None}
+
+    # collect all fragments/features from table
+    fragments_table = set(map(str, table.ids(axis='observation')))
+
+    if len(fragments_table & fragments_tree) <= 0:
+        raise ValueError(('Not a single fragment of your table is part of your'
+                          ' tree. The resulting table would be empty.'))
+
+    def _keep(values, id_, md):
+        return id_ in fragments_table & fragments_tree
+
+    def _remove(values, id_, md):
+        return id_ in fragments_table - fragments_tree
+
+    tbl_positive = table.filter(_keep, axis='observation', inplace=False)
```

Comment:
Much faster to do `tbl_positive = table.filter(fragments_table & fragments_tree, axis='observation', inplace=False)`, and similar for the next line. The filter function allows you to pass in a set of the exact IDs to retain.

Reply:
cool. I changed. But what do you mean by faster. Computationally or code wise?